server/src/controllers/eventController.ts

```diff
@@ -56,70 +57,74 @@ export const eventController = Router();
  *         description: Internal Server Error. An error occurred while processing the request.
  */
 eventController.post('/checkout', async (req: Request, res: Response) => {
-  const {cartItems, formData, donation = 0, discount} = req.body;
-  let orderID = 0;
+  const {cartItems = [], formData, donation = 0, discount} = req.body;
+  let order :orders | null = null;
   let toSend = {id: 'comp'};
   try {
-    if (!cartItems.length && donation === 0) {
+    if (!cartItems.length && !donation) {
       return res.status(400).json({error: 'Cart is empty'});
-    } else if (donation && donation < 0) {
-      return res.status(422).json({error: 'Amount of donation can not be negative'});
     }
+
     if (discount.code != '') {
       await validateDiscount(discount, cartItems, prisma);
     }
 
     const {contactid} = await updateContact(formData, prisma);
-    const {cartRows, orderItems, orderTotal, eventInstanceQueries} =
-      await getOrderItems(cartItems, prisma);
-    const discountAmount = getDiscountAmount(discount, orderTotal);
 
-    const donationItem: LineItem = {
-      price_data: {
-        currency: 'usd',
-        product_data: {
-          name: 'Donation',
-          description: 'A generous donation',
-        },
-        unit_amount: donation * 100,
-      },
-      quantity: 1,
-    };
-    if (donation + orderTotal - discountAmount > 0) {
+    const {
+      ticketCartRows,
+      orderTicketItems,
+      ticketTotal,
+      eventInstanceQueries,
+    } = await getTicketItems(cartItems, prisma);
+
+    const {
+      donations,
+      donationCartRows,
+      donationTotal,
+    } = getDonationItems([donation]);
+
+    const discountAmount = discount.code != ''? getDiscountAmount(discount, ticketTotal): 0;
+    if (ticketTotal + donationTotal - discountAmount > .49) {
       toSend = await createStripeCheckoutSession(
           contactid,
           formData.email,
-          donation,
-        donation ? cartRows.concat(donationItem) : cartRows,
-        orderTotal,
-        discount,
+          ticketCartRows.concat(donationCartRows),
+          {...discount, amountOff: discountAmount},
       );
+    } else if (ticketTotal + donationTotal > 0) {
+      return res.status(400).json({error: 'Cart Total must either be $0.00 USD or greater than $0.49 USD'});
     }
```

Comment:
Do we need a case where the order total is negative?

Reply:
No cart item itself can be negative so I think we might be fine.